src/agent/coverage/src/cobertura.rs

```diff
@@ -0,0 +1,245 @@
+use crate::source::SourceCoverage;
+use crate::source::SourceCoverageLocation;
+use crate::source::SourceFileCoverage;
+use anyhow::Context;
+use anyhow::Error;
+use anyhow::Result;
+use std::time::{SystemTime, UNIX_EPOCH};
+use xml::writer::{EmitterConfig, XmlEvent};
+
+pub fn cobertura(source_coverage: SourceCoverage) -> Result<String, Error> {
+    let mut backing: Vec<u8> = Vec::new();
+    let mut emitter = EmitterConfig::new()
+        .perform_indent(true)
+        .create_writer(&mut backing);
+
+    let unixtime = SystemTime::now()
+        .duration_since(UNIX_EPOCH)
+        .context("system time before unix epoch")?
+        .as_secs();
+
+    emitter.write(
+        XmlEvent::start_element("coverage")
+            .attr("lines-valid", "0")
```

Comment:
For now I kept it as simple as possible with only file, line, and counts- the exact members the SourceCoverage struct has, other things have been hard coded as 0.

Reply:
Makes sense. Before merging, we should explore what happens if we _omit_ the zeroed fields, when it comes to HTML report rendering. To start, we're most interested in the output as rendered in [ADO](https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/test/publish-code-coverage-results?view=azure-devops).